src/Views/FirmwareView.vala

```diff
@@ -143,9 +160,38 @@ public class About.FirmwareView : Gtk.Stack {
                 if (row.device.id == device.id) {
                     update_list.remove (widget);
                 }
+
+                if (row.device.releases.length () > 0 && row.device.latest_release.version != row.device.version) {
+                    num_updatable_devices--;
+                }
             }
         }
 
+        update_row_headers ();
+
         update_list.show_all ();
     }
+
+    private void update_row_headers () {
```

Comment:
This should use https://valadoc.org/gtk+-3.0/Gtk.ListBox.set_header_func.html instead of trying to manually insert headers like this